core/src/main/java/org/apache/accumulo/core/client/admin/ResourceGroupOperations.java

```diff
@@ -0,0 +1,212 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.admin;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.data.ResourceGroupId;
+
+/**
+ * A ResourceGroup is a grouping of Accumulo server processes that have some shared characteristic
+ * that is different than server processes in other resource groups. Examples could be homogeneous
+ * hardware configurations for the server processes in one resource group but different than other
+ * resource groups, or a resource group could be created for physical separation of processing for
+ * table(s).
+ *
+ * A default resource group exists in which all server processes are assigned. The Manager, Monitor,
+ * and GarbageCollector are assigned to the default resource group. Compactor, ScanServer and
+ * TabletServer processes are also assigned to the default resource group, unless their respective
+ * properties are set.
+ *
+ * This object is for defining, interacting, and removing resource group configurations. When the
+ * Accumulo server processes get the system configuration, they will receive a merged view of the
+ * system configuration and applicable resource group configuration, with any property defined in
+ * the resource group configuration given higher priority.
+ *
+ * @since 4.0.0
+ */
+public interface ResourceGroupOperations {
+
+  /**
+   * A method to check if a resource group configuration exists in Accumulo.
+   *
+   * @param group the name of the resource group
+   * @return true if the group exists
+   * @since 4.0.0
+   */
+  boolean exists(String group);
+
+  /**
+   * Retrieve a list of resource groups in Accumulo.
+   *
+   * @return Set of resource groups in accumulo
+   * @since 4.0.0
+   */
+  Set<String> list();
+
+  /**
+   * Create a configuration node in zookeeper for a resource group. If not defined, then processes
+   * running in the resource group will use the values of the properties defined in the system
+   * configuration.
+   *
+   * @param group resource group
+   * @throws AccumuloException if a general error occurs
+   * @throws AccumuloSecurityException if the user does not have permission
+   *
+   * @since 4.0.0
+   */
+  void createConfiguration(final ResourceGroupId group)
+      throws AccumuloException, AccumuloSecurityException;
+
+  /**
+   * Returns properties set for this resource group in zookeeper.
```

Comment:
Should document wether this returns a merged view or not.  May also want to look at some of the other operations interfaces names for consistency.  Like TableOperations have some methods that return merged and non-merged views w/ various names.

Reply:
Fixed in 3fe8748